tests/smoke/support/page/files/allFiles.ts

```diff
@@ -180,4 +193,61 @@ export class AllFilesPage {
     await page.click('#location-picker-btn-confirm')
     await page.goto(startUrl)
   }
+
+  async resourceExist({ 
+    name, 
+    version = false, 
+    notExist = false 
+  }: { 
+    name: string; 
+    version?: boolean; 
+    notExist?: boolean  
+  }): Promise<void> {
+    const { page } = this.actor
+    const startUrl = page.url()
+    const folderPaths = name.split('/')
+    const resouceName = folderPaths.pop()
+
+    if (folderPaths.length) {
+      await cta.files.navigateToFolder({ page: page, path: folderPaths.join('/') })
+    }
+    
+    if (notExist) {
+      const resourceExists = await cta.files.resourceExists({
+        page: page,
+        name: resouceName
+      })      
+      if (resourceExists) throw new Error(`selector was find: "${resouceName}"`)
+    } else if (version) {
```

Comment:
i think it's not good to have all this in one method (increases complexity), can we have another func `checkVersionCount` or similar?

Reply:
created a new method that checks the number of versions